numba_cuda/numba/cuda/models.py

```diff
@@ -42,3 +43,35 @@ def __init__(self, dmm, fe_type):
 
 
 register_model(CUDADispatcher)(models.OpaqueModel)
+
+
+def _as_bfloat(value):
+    # Step 1: Convert to float
+    f = ir.types._as_float(value)
+    # Step 2: Truncate (or round, we choose truncate) last 16 bits
+    bf = f >> 16
+    return bf
+
+
+class BfloatType(ir.types._BaseFloatType):
```

Comment:
Should this be:

```suggestion
class Bfloat16Type(ir.types._BaseFloatType):
```

I don't believe there are other bitwidths, but would this be more consistent naming? Or is there some other subtlety here?

Reply:
I was thinking largely due to the native type in llvm for bf16 is `bfloat`. Trying to be consistent with llvm.

Reply:
Ah, makes sense!